### PR TITLE
Bombchu bowling OK

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Decompiled all functions, and named functions and struct variables in
BgBowlWall
EnBomBowlPit
EnBomBowlMan
EnWallTubo
aside from one function in EnBomBowlMan, and some camera variables in EnBomBowlPit whose purpose I'm not sure about. I'm sure a few of the names can be improved, though.

### CODE REVIEW

**Review by fig02 — APPROVED** [review]
skelanime updated and review changes implemented

File: src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.h [hunk 1]
```diff
@@ -3,12 +3,20 @@
...
 struct EnWallTubo;
 
+typedef void (*EnWallTuboActionFunc)(struct EnWallTubo*, GlobalContext*);
+
 typedef struct EnWallTubo {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x24];
+    /* 0x014C */ EnWallTuboActionFunc actionFunc;
+    /* 0x0150 */ s16 timer;
+    // /* 0x0152 */ char unk_152[0x2];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.h:16]:
```suggestion
```

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.h [hunk 2]
```diff
@@ -8,7 +8,10 @@ struct EnExItem;
 
 typedef struct EnExItem {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x38];
+    /* 0x014C */ char unk_14C[0x14];
+    /* 0x0160 */ s16 unk_160;
+    // /* 0x0164 */ u8 unk_164;
+    /* 0x0165 */ char unk_165[0x20];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.h:14]:
```suggestion
    /* 0x0162 */ char unk_162[0x22];
```

File: src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.h [hunk 3]
```diff
@@ -3,12 +3,38 @@
...
+    /* 0x0150 */ u16 unk_150;
+    /* 0x0152 */ s16 camId;
+    /* 0x0154 */ s16 unk_154;
+    /* 0x0156 */ s16 unk_156; // Set by EnExItem?
+    /* 0x0158 */ s16 timer;
+    /* 0x015A */ s16 prizeIndex;
+    /* 0x015C */ s16 start;
+    // /* 0x015E */ char unk_15E[0x2];
+    /* 0x0160 */ s32 getItemId;
+    // /* 0x0162 */ char unk_162[0x2];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.h:24]:
```suggestion
    /* 0x0160 */ s32 getItemId;
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 4]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/BgBowlWall_Destroy.s")
+static Vec3f sBullseyeOffset[] = {
+    { 0.0f, 210.0f, -20.0f },
+    { 0.0f, 170.0f, -20.0f },
+    { -170.0f, 0.0f, -20.0f },
+    { 170.0f, 0.0f, -20.0f },
+};
+
+static s16 sTargetRot[] = { 0x0000, 0x0000, 0x3FFF, 0xC001 };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:51]:
```suggestion
static s16 sTargetRot[] = { 0x0000, 0x0000, 0x3FFF, -0x3FFF };
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 5]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+    type = this->dyna.actor.params;
+    if (type != 0) {
+        type += (s16)Math_Rand_ZeroFloat(2.99f);
+        this->dyna.actor.shape.rot.z = this->dyna.actor.posRot.rot.z = sTargetRot[type];
+        osSyncPrintf("\n\n");
+    }
+    this->bullseyeCenter.x = sBullseyeOffset[type].x + this->dyna.actor.posRot.pos.x;
+    this->bullseyeCenter.y = sBullseyeOffset[type].y + this->dyna.actor.posRot.pos.y;
+    this->bullseyeCenter.z = sBullseyeOffset[type].z + this->dyna.actor.posRot.pos.z;
+    if (1) {};
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:97]:
```suggestion
    if (1) {}
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 6]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+    }
+}
+
+void BgBowlWall_FallDoEffects(BgBowlWall* this, GlobalContext* globalCtx) {
+    s16 pad;
+    Vec3f effectAccel = { 0.0f, 0.1f, 0.0f };
+    Vec3f effectVelocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f effectPos;
+    s16 quakeIndex;
+    bool wallFallen;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:133]:
```suggestion
    u32 wallFallen;
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 7]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+        for (i = 0; i < 15; i++) {
+            effectPos.x = Math_Rand_CenteredFloat(300.0f) + this->bullseyeCenter.x;
+            effectPos.y = -100.0f;
+            effectPos.z = Math_Rand_CenteredFloat(400.0f) + this->bullseyeCenter.z;
+            EffectSsBomb2_SpawnLayered(globalCtx, &effectPos, &effectVelocity, &effectAccel, 100, 30);
+            effectPos.y = -50.0f;
+            EffectSsHahen_SpawnBurst(globalCtx, &effectPos, 10.0f, 0, 50, 15, 3, HAHEN_OBJECT_DEFAULT, 10, NULL);
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_IT_BOMB_EXPLOSION);
+        }
+        quakeIndex = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:161]:
```suggestion
        quakeIndex = Quake_Add(ACTIVE_CAM, 1);
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 8]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+    Vec3f effectVelocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f effectPos;
+    s16 quakeIndex;
+    bool wallFallen;
+    s32 i;
+
+    wallFallen = false;
+
+    if (this->dyna.actor.params == 0) { // wall collapses backwards
+        Math_SmoothScaleMaxMinS(&this->dyna.actor.shape.rot.x, (u16)-0x3E80, 3, 500, 0);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:139]:
```suggestion
        Math_SmoothScaleMaxMinS(&this->dyna.actor.shape.rot.x, -0x3E80, 3, 500, 0);
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 9]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+    GlobalContext* globalCtx2 = globalCtx;
+    BgBowlWall* this = THIS;
+    u32 frames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F718.s")
+    OPEN_DISPS(globalCtx2->state.gfxCtx, "../z_bg_bowl_wall.c", 441);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F7F8.s")
+    func_80093D84(globalCtx2->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 8,
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:214]:
```suggestion
    gSPSegment(POLY_OPA_DISP++, 0x08,
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c [hunk 10]
```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
...
+               Gfx_TexScroll(globalCtx2->state.gfxCtx, 0, -2 * (frames = globalCtx2->state.frames), 16, 16));
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:215]: Any chance this assignment can be done separately?

File: src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c [hunk 11]
```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
...
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        this->frameCount = (f32)SkelAnime_GetFrameCount(&D_060072AC);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060072AC, 1.0f, 0.0f, this->frameCount, 0, -10.0f);
+        this->eyeMode = CHU_GIRL_EYES_AWAKE;
+        this->blinkTimer = (s16)Math_Rand_ZeroFloat(60.0f) + 20;
+        // Check for beaten Dodongo's Cavern
+        if (!(gSaveContext.eventChkInf[2] & 0x20) && !BREG(2)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c:168]: I suggest having it like this for consistency with the other check
```suggestion
        if (!((gSaveContext.eventChkInf[2] & 0x20) || BREG(2))) {
```

File: src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c [hunk 12]
```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
...
+    this->gameResult = 0;
+
+    if (this->bowlPit != 0) {
+        if ((this->wallStatus[0] != 1) && (this->wallStatus[1] != 1) && (this->bowlPit->status == 2)) {
+            this->gameResult = 1; // Won
+            this->bowlPit->status = 0;
+            // Center HIT!
+            osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 中央ＨＩＴ！！！！ ☆☆☆☆☆ \n" VT_RST);
+        }
+        if ((globalCtx->bombchuBowlingAmmo == -1) && (globalCtx->actorCtx.actorList[3].length == 0) &&
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c:240]:
```suggestion
        if ((globalCtx->bombchuBowlingAmmo == -1) && (globalCtx->actorCtx.actorList[ACTORTYPE_EXPLOSIVES].length == 0) &&
```

File: src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c [hunk 13]
```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
...
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0: // Yes
+                if (gSaveContext.rupees >= 30) {
+                    Rupees_ChangeBy(-30);
+                    this->wallStatus[1] = 0;
+                    this->minigamePlayStatus = 1;
+                    this->wallStatus[0] = this->wallStatus[1];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c:291]: Could this match? Might have to play with the order a bit
```suggestion
                    this->wallStatus[1] = this->wallStatus[1] = 0;
                    this->minigamePlayStatus = 1;
```

File: src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c [hunk 14]
```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
...
+        this->prizeRevealTimer = 10;
+        this->actionFunc = EnBomBowMan_ChooseShowPrize;
+    }
+}
+
+static Vec3f sPrizePosOffset[] = {
+    { 0.0f, 22.0f, 0.0f }, { 0.0f, 22.0f, 0.0f }, { 0.0f, 8.0f, 0.0f }, { 0.0f, 9.0f, 0.0f }, { 0.0f, -2.0f, 0.0f },
+};
+
+static s16 sPrizeRot[] = { 0x4268, 0x4268, 0xFC18, 0x0000, 0x4268, 0x0000 };
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c:372]:
```suggestion
static s16 sPrizeRot[] = { 0x4268, 0x4268, -0x03E8, 0x0000, 0x4268, 0x0000 };
```
Note: All these values make a lot more sense as decimal, like it's often the case, but the whole hex vs dec debate still needs to be resolved separately

File: src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.c [hunk 15]
```diff
@@ -26,15 +33,124 @@ const ActorInit En_Wall_Tubo_InitVars = {
...
+                chuPosDiff.z = chu->actor.posRot.pos.z - this->actor.posRot.pos.z;
+
+                if (((fabsf(chuPosDiff.x) < 40.0f) || (BREG(2))) && ((fabsf(chuPosDiff.y) < 40.0f) || (BREG(2))) &&
+                    (fabsf(chuPosDiff.z) < 40.0f || (BREG(2)))) {
+                    this->chuGirl->wallStatus[this->actor.params] = 1;
+                    chu->unk_150 = 2;
+                    func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+                    this->timer = 60;
+                    EffectSsBomb2_SpawnLayered(globalCtx, &this->explosionCenter, &effVelocity, &effAccel, 200, 40);
+                    quakeIndex = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.c:96]:
```suggestion
                    quakeIndex = Quake_Add(ACTIVE_CAM, 1);
```

File: src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.c [hunk 16]
```diff
@@ -20,25 +32,191 @@ const ActorInit En_Bom_Bowl_Pit_InitVars = {
...
+                func_8010B680(globalCtx, this->actor.textId, NULL);
+                this->unk_154 = 5;
+                func_80078884(NA_SE_EV_HIT_SOUND);
+                func_8002DF54(globalCtx, NULL, 8);
+                this->status = 1;
+                this->actionFunc = EnBomBowlPit_CameraDollyIn;
+                break;
+            } else {
+                chu = (EnBomChu*)chu->actor.next;
+            }
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.c:116]: no else here imo, if possible

File: src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.c [hunk 17]
```diff
@@ -20,25 +32,191 @@ const ActorInit En_Bom_Bowl_Pit_InitVars = {
...
+}
+
+void EnBomBowlPit_SetupGivePrize(EnBomBowlPit* this, GlobalContext* globalCtx) {
+    if (this->unk_156 != 0) {
+        switch (this->prizeIndex) {
+            case 0:
+                gSaveContext.itemGetInf[1] |= 2;
+                break;
+            case 1:
+                gSaveContext.itemGetInf[1] |= 4;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bom_Bowl_Pit/z_en_bom_bowl_pit.c:164]:
```suggestion
                gSaveContext.itemGetInf[1] |= 4;
                break;
```

File: src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.c [hunk 18]
```diff
@@ -26,15 +33,124 @@ const ActorInit En_Wall_Tubo_InitVars = {
...
+        EffectSsBomb2_SpawnLayered(globalCtx, &effPos, &effVelocity, &effAccel, 100, 30);
+        EffectSsHahen_SpawnBurst(globalCtx, &effPos, 10.0f, 0, 50, 15, 3, HAHEN_OBJECT_DEFAULT, 10, NULL);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+    }
+
+    if (this->timer == 0) {
+        wall = (BgBowlWall*)this->actor.parent;
+
+        if ((wall != NULL) && (wall->dyna.actor.update != NULL)) {
+            wall->isHit = 1;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Wall_Tubo/z_en_wall_tubo.c:129]:
```suggestion
            wall->isHit = true;
```

File: src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c (hunk 10 above)
Comment (fig02) [reply src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c:215]: messed with it a bit just now, cant get anything to work. tried some usual tricks like * 1 and moving the negative sign and stuff

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**